pkg/ibmvpc/vpc.go

```diff
@@ -753,6 +780,111 @@ func (tgw *TransitGateway) AllowedConnectivity(src, dst vpcmodel.VPCResourceIntf
+	// array being applied first, and those at the end of the array is applied last, or just before the default.
+	resStr := fmt.Sprintf("transit-connection: %s", *prefix.tc.Name)
+	if prefix.index == -1 { // default
+		defaultStr, err := prefixDefaultStr(prefix.tc)
+		if err != nil {
+			return "", err
+		}
+		return resStr + defaultStr, nil
+	}
+	if len(prefix.tc.PrefixFilters) < prefix.index+1 {
+		return "", fmt.Errorf("np-guard error: prefix index %d does not exists in transit connection %s of transit gateway %s",
+			prefix.index, *prefix.tc.Name, tgw.Name())
+	}
+	prefixFilter := prefix.tc.PrefixFilters[prefix.index]
+	actionName, err := actionNameStr(prefixFilter.Action)
+	if err != nil {
+		return "", err
+	}
+	resStr += fmt.Sprintf(", index: %v, action: %s", prefix.index, actionName)
+	if prefixFilter.Ge != nil {
+		resStr += fmt.Sprintf(", ge: %v", *prefixFilter.Ge)
+	}
+	if prefixFilter.Le != nil {
+		resStr += fmt.Sprintf(", le: %v", *prefixFilter.Le)
+	}
+	resStr += fmt.Sprintf(", prefix: %s", *prefixFilter.Prefix)
+	return resStr, nil
+}
+
+// for an action of type *string as stored in *datamodel.TransitConnection returns allow/deny
+func actionNameStr(action *string) (string, error) {
+	actionBool, err := parseActionString(action)
+	if err != nil {
+		return "", err
+	}
+	if actionBool {
+		return "allow", nil
+	}
+	return "deny", nil
+}
+
+func (tgw *TransitGateway) StringPrefixDetails(src, dst vpcmodel.Node, verbose bool) (string, error) {
+	prefix := tgw.prefixOfSrcDst(src, dst)
+	transitEnablesConn := vpcmodel.HasNode(tgw.sourceNodes, src) && vpcmodel.HasNode(tgw.destNodes, dst)
+	if verbose {
+		tgwRouterFilterDetails, err := tgw.tgwPrefixStr(*prefix)
+		if err != nil {
+			return "", err
+		}
+		action := "blocks"
+		if transitEnablesConn {
+			action = "allows"
+		}
+		return fmt.Sprintf("transit gateway %s %s connection with the following prefix\n\t%s\n\n",
+			tgw.Name(), action, tgwRouterFilterDetails), nil
+	}
+	noVerboseStr := fmt.Sprintf("cross-vpc-connection: transit-connection %s of transit-gateway %s ", tgw.Name(), *prefix.tc.Name)
+	if transitEnablesConn {
+		return noVerboseStr + "allows connection", nil
+	}
+	return noVerboseStr + "denys connection", nil
+}
+
+func (tgw *TransitGateway) prefixOfSrcDst(src, dst vpcmodel.Node) *tgwPrefix {
+	// <src, dst> routed by tgw given that source is in the tgw,
+	// and there is a prefix defined for the dst,
+	// the relevant prefix is determined by match of the ap the dest's node is in (including default)
+	if vpcmodel.HasNode(tgw.sourceNodes, src) {
+		for routeCIDR, prefix := range tgw.vpcApsPrefixes[dst.VPC().UID()] {
+			if dst.IPBlock().ContainedIn(routeCIDR) {
+				return &prefix
+			}
+		}
+	}
+	return nil
+}
+
+func (fip *FloatingIP) StringPrefixDetails(src, dst vpcmodel.Node, verbose bool) (string, error) {
+	return "", nil
+}
+
+func (pgw *PublicGateway) StringPrefixDetails(src, dst vpcmodel.Node, verbose bool) (string, error) {
+	return "", nil
+}
```

Comment:
put these  two methods in the right place

Reply:
You mean with the other class functions in the same file? Done
Did you mean something else? what?